clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -2328,6 +2328,116 @@ test_replay_timestamps()
 #endif // HAS_ZIP
 }
 
+#ifdef HAS_ZIP
+// We subclass scheduler_t to access its record struct and functions.
+class test_noeof_scheduler_t : public scheduler_t {
+public:
+    void
+    write_test_schedule(std::string record_fname)
+    {
+        // We duplicate test_scheduler_t but we have one input ending early before
+        // eof.
+        scheduler_t scheduler;
+        std::vector<schedule_record_t> sched0;
+        sched0.emplace_back(scheduler_t::schedule_record_t::VERSION, 0, 0, 0, 0);
+        sched0.emplace_back(scheduler_t::schedule_record_t::DEFAULT, 0, 0, 4, 11);
+        // There is a huge time gap here.
+        sched0.emplace_back(scheduler_t::schedule_record_t::DEFAULT, 2, 7,
```

Comment:
Re-iterate in comment here that "Max numeric value means continue until EOF."

Reply:
Done.